File: meetupchief/src/main/java/pl/mobilewarsaw/meetupchief/dagger/module/DatabaseModule.java
package pl.mobilewarsaw.meetupchief.dagger.module;


import android.content.Context;

import pl.mobilewarsaw.meetupchief.database.Database;

public class DatabaseModule {

    private Context context;

    public DatabaseModule(Context context) {
        this.context = context;
    }

    Database provideDataBase() {
        return Database.getInstance(context);
    }
}

File: meetupchief/src/main/kotlin/pl/mobilewarsaw/meetupchief/config/injekt/module/DatabaseModule.kt
package pl.mobilewarsaw.meetupchief.config.injekt.module

import android.content.Context
import pl.mobilewarsaw.meetupchief.database.Database
import uy.kohesive.injekt.api.InjektModule
import uy.kohesive.injekt.api.InjektRegistrar
import uy.kohesive.injekt.api.fullType


class DatabaseModule(val context: Context) : InjektModule {

    override fun InjektRegistrar.registerInjectables() {
        addFactory(fullType<Database>()) { Database.getInstance(context) }
    }
}